common/src/main/java/earth/terrarium/olympus/client/components/Widgets.java

```diff
@@ -212,6 +212,55 @@ public static TextBox intInput(State<Integer> state) {
         return intInput(state, Consumers.nop());
     }
 
+    public static TextBox doubleInput(State<Double> state, Consumer<TextBox> factory) {
```

Comment:
this parsing code should be put into a util and always return a double and then map the state for the int to truncate